Formula/t/teslamate.rb

```diff
@@ -0,0 +1,119 @@
+    system "npm", "install", "--prefix", "./assets", *std_npm_args(prefix: false)
+    system "npm", "run", "deploy", "--prefix", "./assets"
+
+    with_env("MIX_ENV" => "prod") do
+      system "mix", "do", "phx.digest,", "release", "--overwrite"
+    end
+
+    touch buildpath/"teslamate.env"
+    etc.install "teslamate.env"
+    libexec.install Dir["_build/prod/rel/teslamate/*"]
+    bin.install_symlink Dir["#{libexec}/bin/teslamate"]
+
+    # Corresponds to https://github.com/teslamate-org/teslamate/blob/main/entrypoint.sh
+    (bin/"teslamate_brew_services").write <<~EOS
+      #!/bin/bash
+      set -e
+
+      source #{etc}/teslamate.env
+
+      : "${DATABASE_HOST:="127.0.0.1"}"
+      : "${DATABASE_PORT:=5432}"
+
+      # wait until Postgres is ready
+      if [[ -n "${DATABASE_SOCKET_DIR}" ]]; then
+        # Check for Unix socket
+        while [ ! -S "${DATABASE_SOCKET_DIR}/.s.PGSQL.${DATABASE_PORT}" ]; do
+          echo waiting for postgres socket at "${DATABASE_SOCKET_DIR}/.s.PGSQL.${DATABASE_PORT}"
+          sleep 1s
+        done
+      else
+        # Check for TCP connection
+        while ! nc -z "${DATABASE_HOST}" "${DATABASE_PORT}" 2>/dev/null; do
+          echo waiting for postgres at "${DATABASE_HOST}":"${DATABASE_PORT}"
+          sleep 1s
+        done
+      fi
+
+      # apply migrations
+      #{bin}/teslamate eval "TeslaMate.Release.migrate"
+
+      # start the application
+      exec #{bin}/teslamate start
+    EOS
+  end
+
+  service do
+    run opt_bin/"teslamate_brew_services"
+    keep_alive true
+    log_path var/"log/teslamate.log"
+    error_log_path var/"log/teslamate.log"
+    working_dir var
+  end
+
+  test do
+    ENV["LC_ALL"] = "C"
+
+    pg_port = free_port
+    pg_bin = Formula["postgresql@17"].opt_bin
+    pg_ctl = pg_bin/"pg_ctl"
+    datadir = testpath/"postgres"
+    system pg_ctl, "init", "-D", datadir
+
+    (datadir/"postgresql.conf").write <<~EOS, mode: "a+"
+      port = #{pg_port}
+      unix_socket_directories = '#{datadir}'
+    EOS
+
+    system pg_ctl, "start", "-D", datadir, "-l", testpath/"postgres.log"
+    begin
+      system pg_bin/"createdb", "-h", datadir, "-p", pg_port.to_s, "teslamate"
+      system pg_bin/"createuser", "-h", datadir, "-p", pg_port.to_s, "-s", "teslamate"
+
+      # Run Teslamate with the test database
+      ENV["DATABASE_USER"] = "teslamate"
+      ENV["DATABASE_PASS"] = ""
+      ENV["DATABASE_NAME"] = "teslamate"
+      ENV["DATABASE_HOST"] = "127.0.0.1"
+      ENV["DATABASE_PORT"] = pg_port.to_s
+      ENV["DISABLE_MQTT"] = "true"
+      require "pty"
+      PTY.spawn(opt_bin/"teslamate_brew_services") do |r, _w, pid|
+        require "timeout"
+        found_match = false
+        Timeout.timeout(30) do
+          r.each do |line|
+            if line.include?("Access TeslaMateWeb.Endpoint at http://localhost")
+              found_match = true
+              break
+            end
+          end
+        end
```

Comment:
Why not just sleep?

Reply:
The goal here was to stream the process output until a certain marker is seen or a timeout is reached (in which case the test fails). What did you have have in mind?

Reply:
Just start it, write output somewhere and check that output after a sleep. It'll be much easier to read.